configs/common/packages.yaml

```diff
@@ -72,232 +65,196 @@
-    nlohmann-json-schema-validator:
-      version: ['2.1.0']
     odc:
-      version: ['1.4.6']
-      variants: ~fortran
+      require: '@1.4.6 ~fortran'
     openblas:
-      version: ['0.3.24']
-      variants: +noavx512
+      require: '@0.3.24 +noavx512'
     openmpi:
-      variants: ~internal-hwloc +two_level_namespace
-    # Pin openssl to avoid duplicate packages being built
+      require: '~internal-hwloc +two_level_namespace'
     openssl:
-      variants: +shared
+      require: '+shared'
     p4est:
-      require: "@2.8"
+      require: '@2.8'
     parallelio:
-      version: ['2.6.2']
-      variants: +pnetcdf
+      require: '@2.6.2 +pnetcdf'
     parallel-netcdf:
-      version: ['1.12.3']
+      require: '@1.12.3'
     pflogger:
-      #version: ['1.12.0']
-      variants: +mpi
+      require: '+mpi'
     pixman:
-      variants: +pic
-    # Do not build pkgconf - https://github.com/jcsda/spack-stack/issues/123
-    pkgconf:
-      buildable: False
+      require: '+pic'
     prod-util:
-      version: ['2.1.1']
+      require: '@2.1.1'
     proj:
-      version: ['8.1.0']
-      variants: ~tiff
+      require: '~tiff'
     python:
-      require: "@3.10.13"
+      require: '@3.10.13'
     py-attrs:
       # https://github.com/JCSDA/spack-stack/issues/740
-      version: ['21.4.0']
+      require: '@21.4.0'
     py-cartopy:
-      variants: +plotting
-      require: "@0.21.1"
+      require: '+plotting'
     py-cryptography:
-      variants: +rust_bootstrap
+      require: '+rust_bootstrap'
     # Introduced in https://github.com/JCSDA/spack-stack/pull/894, pin py-cython
     # to avoid duplicate packages being built (cylc dependencies soft-want @3:)
     py-cython:
-      require: "@0.29.36"
-    ## https://github.com/JCSDA/spack-stack/issues/980
-    #py-gitpython:
-    #  require: "@3.1.27"
-    py-h5py:
-      version: ['3.7.0']
-      variants: ~mpi
+      require: '@0.29.36'
     # To avoid duplicate packages
     py-flit-core:
-      require: "@3.8.0"
+      require: '@3.8.0'
+    py-h5py:
+      require: '~mpi'
     # To avoid duplicate packages
     py-jinja2:
-      require: "@3.0.3"
+      require: '@3.0.3'
+    py-meson-python:
+       require: '@0.15.0'
     # Comment out for now until build problems are solved
     # https://github.com/jcsda/spack-stack/issues/522
     # see also ewok-env virtual package and container
     # README.md
     #py-mysql-connector-python:
-    #  version: ['8.0.32']
+    #  require: '@8.0.32'
     py-netcdf4:
-      version: ['1.5.8']
-      variants: ~mpi
+      require: '@1.5.8 ~mpi'
+    # py-numpy@1.26 causes many build problems with older Python packages
+    # also check Nautilus site config when making changes here
     py-numpy:
-      require: ['@1.22.3']
+      require: '@:1.25'
     py-pandas:
-      variants: +excel
-    # Pin py-poetry-core to avoid duplicate Python packages
-    py-poetry-core:
-      require: '@1.8.1'
+      require: '+excel'
```

Comment:
We still need to pin the py-pybind11 to avoid duplicates on Hera when compiling with %gcc:
```
    py-pybind11:
      require: '@2.11.0'
```

Reply:
Sigh, ok. Let me add that back in. Is that because of the old gcc 9.2.0?

Reply:
Yes-- I tried concretizing with 13.2.0 without the pinned py-pybind11 version and it worked fine with no duplicates.